### DISCUSSION TITLE
How to implement regenerate-code

### DISCUSSION THREAD

**Discussion by KyrieCui** [microsoft/autogen#942] [Q&A] [answered]
I noticed that if I give a task about generate Python Script, if execute the generated code will have some bug, the agent will solve the code automatically. I want to ask how to implement this in the framework? Since I read the source code but I cannot find when the code run fail, how to give the error msg for the agent to improve it.

**Comment by rickyloynd-microsoft** [comment]
Are you referring to the type of example shown on the README? (plot NVDA vs. TESLA)

**Reply by KyrieCui** [reply]
Hello Sir. 
Thank you for your reply. I am just curious about how the agent regenerate code after executing the generated code and returning some error msg. For example, when I ask the agent to generate Python code for an algorithm, it may not execute successfully once and return some error message with exit code :1. After that, the agent seems to use this error message to regenerate the code automatically and finally give me the right answer.  Like the code in agentchat_auto_feedback_from_code_execution.ipynb

**Reply by rickyloynd-microsoft** [reply, marked as answer]
I think you're on the right track. For instance, if a user asks chatgpt to write some code, it's expected that the user will then run the code themselves. And if an error message is produced, the user can paste that error message into the chat window. Then chatgpt will often rewrite the code to resolve the error. Similarly, in autogen, the UserProxyAgent has the ability to execute code and send error messages back to the AssistantAgent that wrote the code, so that the assistant can reply with a (hopefully) corrected version of the code to be tried next. Does that answer your question, or is there some part of it that you would like to understand better?

**Reply by KyrieCui** [reply]
Yep. Thank u for your answer and it helps me a lot to understand how the framework work.
